test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -923,12 +926,16 @@ def _check_common_mock_calls_batch(mocks, reqs1, reqs2, parent_wsid):
     #     ]
     # )
 
-    final_expected_parent_job = Job()
-    final_expected_parent_job.id = ObjectId(_JOB_ID)
-    final_expected_parent_job.user = _USER
-    final_expected_parent_job.child_jobs = [_JOB_ID_1, _JOB_ID_2]
-    final_got_parent_job = sdkmr.save_job.call_args_list[4][0][0]
-    assert_jobs_equal(final_got_parent_job, final_expected_parent_job)
+    # Test to see if add_child jobs is called with correct batch_container and children
+    expected_batch_container = Job()
+    expected_batch_container.id = ObjectId(_JOB_ID)
+    expected_batch_container.user = _USER
+
+    batch_job = sdkmr.add_child_jobs.call_args_list[0][1]["batch_job"]
+    sdkmr.add_child_jobs.assert_called_once_with(
+        batch_job=expected_batch_container, child_jobs=[_JOB_ID_1, _JOB_ID_2]
```

Comment:
This relies on equality, which doesn't work for `Job` objects. You'll have to pull the `call_args_list` apart and test the parts individually, unfortunately.

Reply:
Buuut tests are passing... huh that's weird

Reply:
Ok, I don't see how this is passing, not only from the equality issue, but also because the batch jobs have a bunch more fields than just `id` and `user`

Reply:
Oh, I bet it's because they have the same object ID, which is what `Job` equality is based on. So this test doesn't actually check that the call is corrent, but the `assert_jobs_equal` line below does. So this is ok as is.

Reply:
Actually if you could add a comment here explaining that the assert below is necessary because of how equality works for `Job` objects and that the assert_called_once_with doesn't correctly check the job object that'd help the next person that comes along

Reply:
Added

Reply:
Looks like there's some sort of interpolation in the comment?
```
Job objects
    (  print(jobs_to_update) because 
```

Reply:
fixed

Reply:
:+1: